tests/foreman/cli/test_hostgroup.py

```diff
@@ -38,28 +38,46 @@
 from robottelo.cli.factory import make_subnet
 from robottelo.cli.hostgroup import HostGroup
 from robottelo.cli.proxy import Proxy
+from robottelo.cli.puppet import Puppet
 from robottelo.config import settings
 from robottelo.datafactory import invalid_id_list
 from robottelo.datafactory import invalid_values_list
 from robottelo.datafactory import parametrized
 from robottelo.datafactory import valid_hostgroups_list
 
-
 pytestmark = [
     pytest.mark.skipif(
         (not settings.robottelo.REPOS_HOSTING_URL), reason='Missing repos_hosting_url'
     )
 ]
-PUPPET_MODULES = [
-    {'author': 'robottelo', 'name': 'generic_1'},
-    {'author': 'robottelo', 'name': 'generic_2'},
-]
+
+pc_name = 'generic_1'
 
 
 @pytest.fixture(scope='module')
-def content_source(default_sat):
+def env(module_org, module_location, default_sat):
+    """Return the puppet environment."""
+
+    env_name = default_sat.create_custom_environment(repo=pc_name)
+    env = default_sat.api.Environment().search(query={'search': f'name={env_name}'})[0].read()
+    env = entities.Environment(
+        id=env.id,
+        location=[module_location],
+        organization=[module_org],
+    ).update(['location', 'organization'])
+    return env
+
+
+@pytest.fixture(scope='module')
+def puppet_classes(env):
+    """Return puppet classes."""
+    return [Puppet.info({'name': pc_name, 'puppet-environment': env.name})]
+
+
+@pytest.fixture(scope='module')
+def content_source():
     """Return the proxy."""
-    return Proxy.list({'search': f'url = {default_sat.url}:9090'})[0]
+    return Proxy.list({'search': f'url = https://{settings.server.hostname}:9090'})[0]
```

Comment:
I thought we were supposed tp move away from this `settings.server.hostname` in favour of `default_sat.url`